Graveyard insights: Fixing the issues related to the planet descriptions and mission structuring (#5433)
endless-sky/endless-sky: data/ka'het/ka'het missions.txt

	invisible
	source
		government "Remnant"
		not planet "Viminal"
	to offer
		or
			has "Graveyard Label: offered"

> 			has "Ringworld Debris: offered"

Reviewer: You only got one of the missions for this.

> 			has "Ringworld Debris: offered"

Author: Done

				`	(Yes.)`
				`	(Not now.)`
					defer
			branch Taely

> 				has "remnant met taely"

Reviewer: This mission doesn't offer if you have this condition, so this branch can never execute. I think a better course of action would be to allow Remnant 1B to offer anywhere.
Labels should be lowercase.

Author: Changed not to care about having "remnant met taely" or not, as that is not unnecessary since it has been changed not to offer on Viminal (there is the case of the player landing on viminal without having met taely and not having the mission, but this will pop up as soon as they land on another remnant planet so it doesn't seem problematic to me).
I prefer to still have this as the non-viminal mission and keep 1B only there, as as far as the player is concerned that is the place where Taely is

Reviewer: Now a player who hasn’t met Taely but has found the Graveyard will get no information when landing in Viminal.

Author: Added a third mission for Viminal

			choice
				`	"Wait, why did you call them 'shells' and not ships?"`
			label shells

> 			`	"You see, <first>," she says, "the Ka'het aren't normal spaceships. The debris we found were artificial exoskeletons, built around huge living beings; it's unlikely a species that big could have developed that sort of technology, so we presume someone must have created it for them. We call them the Builders. After we lost the first scout ships in the Graveyard, the prefects at the time decided that we should not stop our attempts to scout the region, but instead ensure to never have to decloak before getting back to safe space. Thanks to those protocols, some of which are still applied today, we were able to extensively explore most of the Graveyard and decipher a good amount of the written language we found. A large number of missions down there were limited to recovering some debrises from a designated site and taking them here before running out of fuel though, so there is still a lot we have yet to discover." You keep talking about the Graveyard with Taely for a while longer, until one of the people she was talking to earlier comes closer and shows her something on a touchpad. "I have to go now, Captain," she tells you. "Good luck."`

Author: The meaning of that phrase was that, while taking a few relics from a site and taking them to a laboratory can be very useful, having time to properly explore the planets and the surroundings of the site can be just as useful.
Still, I'll go with your second suggestion, as that specification seems unnecessary now

				decline
	on decline
		event "Graveyard insights"
		log `Asked Taely about the Graveyard and the Ka'het and learned of the Remnant's previous contacts between them.`
		log "Factions" "Ka'het" `The Ka'het are a species which live in the Graveyard, an area south of the Ember Waste. They are large lifeforms which can travel in space using artificial exoskeletons and will attack anyone on sight, even other Ka'het.`

> mission "Graveyard Insights Compatibility"

Reviewer: Compatibility missions are typically all lowercase.

	landing
	invisible
	to offer
		has "First Contact: Ka'het: Remnant 1B: offered"
		not "event: Graveyard insights"
	on offer
		event "Graveyard insights"
		fail

> event "Graveyard insights"

Reviewer: Events are typically all lowercase.

Author: The game already has event "Poisonwood reverts back to Republic" and event "Tarazed neutrality" in vanilla

Reviewer: Events are still typically lower case.

Author: Done

	planet Alix

> 		spaceport `Long ago, the Builders stripped this moon to build the nearby Parii'het space station and produce Ka'het hulls. The Ka'het attacks were limited to the inhabited settlements on Alix; had they also destroyed the mining bases and the unmanned drones, the station would have run out of supplies thousands of years ago.`

Reviewer: You sure about these? Adding a spaceport means the player is suddenly able to get fuel from landing on these planets just because they got more information from the Remnant.

Author: That's what I found out while testing too, but is there really an option other than completely changing the planetary descriptions and removing the old one? While remaining stranded in the Graveyard is definetly nice, until we have something like https://github.com/endless-sky/endless-sky/issues/666 that separates spaceport and automatic refuel this seems like the best course of action, as it isn't that big of a drawback to have the player refuel there

Reviewer: > While remaining stranded in the Graveyard is definetly nice, until we have something like #666 that separates spaceport and automatic refuel this seems like the best course of action, as it isn't that big of a drawback to have the player refuel there

I think that being able to suddenly refuel in the Graveyard is a very big drawback compared to what is gained by having expanded descriptions in this way.

Author: Changed not to use spaceport but only the normal descriptions

		spaceport `	Incidentally, when the invasion broke out, Osaeli was also the world closest to the Ka'het fleets: it kept being devastated for years, until every Builder on the planet was killed.`
	planet Palarei
		spaceport `Thanks to their earlier exploration vessels, here the Builders encountered the first lifeforms that evolved outside of their home planet. While the climate here was very different from what they were used to, the fruits the local flora produced were popular enough that Palarei became a major farming spot for a long time.`
		spaceport `	The farms on Palarei were one of the few successful attempts at creating fully automatic facilities for the Builders. A few dozen of them, partly thanks to the rudimental repair systems they tried to develop, is still working even today.`
	planet Parii'het
		spaceport `Parii'het was almost finished when the invasion broke out, and already capable of (slowly) creating Ka'het. Thanks to the mining operations on Alix and the occasional asteroid mining operated by the Mar drones, most of the facilities are still fully operational to this day.`

> 		spaceport `	Differently from most of the Graveyard, the station doesn't seem to have ever suffered any damage from the Ka'het, nor from the Builders trying to regain control over it.`

Reviewer 2: ```suggestion
		spaceport `	Different from most of the Graveyard, the station doesn't seem to have ever suffered any damage from the Ka'het, nor from the Builders trying to regain control over it.`
```

	planet Retilie
		spaceport `Retilie was used by the builders mainly for fishing, leading to a lack of infrastructural development. The few settlements on the planet were never as advanced as the ones on the nearby worlds, and when the Ka'het arrived there was basically nothing to stop them: after a few attacks, the planet was uninhabited again.`
		spaceport `	In the oceans, the many species of fish continue to prosper, unaware of the destruction the planet has undergone.`
	planet Seli'het
		spaceport `Seli'het was the only Het station the Builders ever completed, orbiting Giribea. The young Ka'heta are brought here once they reach the "Age of Assembly;" strange, lanky machines start molding an exoskeleton onto their bodies right away. As the Ka'het grow, their exoskeleton is expanded and various outfits are added, transforming them into living weapons.`

> 		spaceport `	In this system the Ka'het behave differently than basically anywhere else as they don't seem to ever attack each other, and some even seem to have started collaborating.`

Reviewer 2: ```suggestion
		spaceport `	In this system, the Ka'het behave differently than basically anywhere else. They don't seem to ever attack each other, and some even seem to have started collaborating.`
```

	planet Siteria

> 		spaceport `Siteria is the Builder homeworld, both the place where their journey began and where it ended. After a dozen millennia of underpopulation due to its closeness to the Quarg ring, the Ka'het invasion forced them to retreat. As one of the few remaining planets capable of sustaining farming and by far the one with the best climate, for a few centuries it had again the highest number of inhabitants of all the Builders planets.`

Reviewer 2: ```suggestion
		spaceport `Siteria is the Builder homeworld, both the place where their journey began and where it ended. After a dozen millennia of underpopulation due to its proximity to the Quarg ring, the Ka'het invasion forced them to retreat. As one of the few remaining planets capable of sustaining farming and by far the one with the best climate, for a few centuries it had again the highest number of inhabitants of all the Builders planets.`
```

		spaceport `	The large number of fields and farms around the planet allowed the Builders to survive for centuries, hiding in sparse strongholds and citadels. Almost three hundred years passed before the Ka'het were able to destroy the last colony.`
	planet Veliante

> 		spaceport `Incidentally, Veliante was one of the worlds that suffered the heaviest damage during the Ka'het invasion. Here one of the earliest attacks hit some sort of advanced nuclear facility, starting a meltdown; the fallout quickly spread across most of the planet, leaving all Builder settlements uninhabitable.`

Reviewer 2: ```suggestion
		spaceport `Veliante was one of the worlds that suffered the heaviest damage during the Ka'het invasion. Here one of the earliest attacks hit some sort of advanced nuclear facility, starting a meltdown; the fallout quickly spread across most of the planet, leaving all Builder settlements uninhabitable.`
```

				decline
	on decline
		event "graveyard insights"
		log `Asked Taely about the Graveyard and the Ka'het and learned of the Remnant's previous contacts between them.`
		log "Factions" "Ka'het" `The Ka'het are a species which live in the Graveyard, an area south of the Ember Waste. They are large lifeforms which can travel in space using artificial exoskeletons and will attack anyone on sight, even other Ka'het.`

mission "graveyard insights compatibility"
	landing
	invisible
	to offer
		has "First Contact: Ka'het: Remnant 1B: offered"
		not "event: Graveyard insights"
	on offer
		event "graveyard insights"
		fail

event "graveyard insights"
	planet Alix

> 		description `	 Here the Ka'het attacks were limited to the inhabited settlements; had they also destroyed the mining bases and the unmanned vessels, station Parii'het would have run out of supplies thousands of years ago.`

Reviewer: These should be `add description`, as otherwise these are overwriting the existing descriptions with what is in this event.

mission "graveyard insights compatibility"
	landing
	invisible
	to offer
		has "First Contact: Ka'het: Remnant 1B: offered"
		not "event: Graveyard insights"
	on offer
		event "graveyard insights"
		fail

event "graveyard insights"
	planet Alix
		description `	 Here the Ka'het attacks were limited to the inhabited settlements; had they also destroyed the mining bases and the unmanned vessels, station Parii'het would have run out of supplies thousands of years ago.`
	planet Celeatis
		description `	This planet appears to have been home to one of the largest Builder research centers, which focused on studying the anomaly in the system. Had they found out how to stabilize it and use it as a wormhole, perhaps their species would have survived and spread to the rest of the Ember Waste.`
	planet Ceilia'sei
		description `	The main purpose of Celia'sei to act like some sort of anti-missile platform. But when the Ka'het invasion broke out for the second time, it had no way to defend deadly Ravager Beams, and the station's defense systems were subsequently lost forever.`
	planet Eavine

> 		description `	This moon was also home for the second Ka'het wave, refusing to obey again and for the last time. This moon had been terraformed to make it a more appealing place to live, but after their civilization collapsed the moon began returned to its natural state. When it reverted to its original sub-zero temperatures, the facilities stopped working, and the Ka'het production in this site came to an end.`

Reviewer 2: ```suggestion
		description `	This moon was also home for the second Ka'het wave, which, same as the first, refused to obey their creators. It had been terraformed to make it a more appealing place to live, but after their civilization collapsed the moon began returned to its natural state. When it reverted to its original sub-zero temperatures, the facilities stopped working, and the Ka'het production in this site came to an end.`
```
I'll be honest: I don't remember all of the lore of the Builders and the Ka'het off the top of my head, so hopefully this change is still historically correct. Let me know if it isn't.

Author: It is

	planet "Letis'sei"

> 		description `	As it was designed to be crewed as minimally as possible, only a tiny fraction of the station was inhabitable. The rest of the station, instead, was dedicated to advanced computing and targeting systems, most of which are still intact today.`

Reviewer 2: ```suggestion
		description `	As it was designed to be crewed as minimally as possible, only a tiny fraction of Letis'sei station was inhabitable. The rest of the station was instead dedicated to advanced computing and targeting systems, most of which are still intact today.`
```
Your original is fine, these are just suggested for clarification/flow.

	planet Osaeli

> 		description `	For most of the thousands of years that passed between the time their discovery of spaceflight and their extinction, Osaeli was the de facto capital of the Builders, a key part of their economy and society. The first Het station was constructed orbiting this planet's star, with the purpose of protecting them from any external threat.`

Reviewer 2: ```suggestion
		description `	For most of the thousands of years that passed between the time their discovery of spaceflight and their extinction, Osaeli was the de facto capital of the Builders, a key part of their economy and society. The first Het station was constructed orbiting this planet's star with the purpose of protecting them from any external threat.`
```

	planet Palarei
		description `	Thanks to their earlier exploration vessels, this was where the Builders encountered the first lifeforms that evolved outside of their home planet. While the climate was drastically different for the Builders, the fruits the local flora produced were popular enough that Palarei became a major farming planet.`
	planet Parii'het

> 		description `	Parii'het was almost finished when the invasion broke out, and already capable of creating Ka'het. Thanks to the mining operations on Alix and the occasional asteroid mining operated by the Mar unmanned vessels, most of the facilities are still fully operational to this day.`

Reviewer 2: ```suggestion
		description `	Parii'het was almost finished when the invasion broke out, and already capable of creating Ka'het. Thanks to the mining operations on Alix and the occasional asteroid mining by the Mar unmanned vessels, most of the facilities are still fully operational to this day.`
```
OR
```suggestion
		description `	Parii'het was almost finished when the invasion broke out, and already capable of creating Ka'het. Thanks to the mining on Alix and the occasional asteroid mining operated by the Mar unmanned vessels, most of the facilities are still fully operational to this day.`
```
Think I missed this the first pass. Just wanted to remove the double usage of `operation`. This is personal preference, and if you wanted to either keep as-is or substitute another word instead of removing one, that would work too.

	planet Retilie

> 		description `	As it was mainly used for fishing, the settlements on this planet were never as advanced as the ones on the nearby worlds, and when the Ka'het arrived there was basically nothing to stop them: after a few attacks, the planet was uninhabited again.`

Reviewer 2: ```suggestion
		description `	As it was mainly used for fishing, the settlements on this planet were never as advanced as the ones on nearby worlds, so when the Ka'het arrived, there was basically nothing to stop them: after a few attacks, the planet was uninhabited again.`
```
OR
```suggestion
		description `	As it was mainly used for fishing, the settlements on this planet were never as advanced as the ones on the nearby worlds, and when the Ka'het arrived there was virtually nothing to stop them: after a few attacks, the planet was uninhabited again.`
```
`Basically` vs `virtually` is personal preference. Wanted to make two suggestions since there were grammar fixes that needed to happen regardless of whether you preferred that word choice.

	planet Seli'het
		description `	This was the only Het station the Builders ever completed, orbiting Giribea. As the young Ka'heta reach their "Age of Assembly," strange, lanky machines start molding an exoskeleton onto their bodies right away. As the Ka'heta grow, their exoskeleton is expanded and various outfits are added, transforming them into living weapons.`
	planet Siteria

> 		description `	This was the Builder homeworld, both the place where their journey began and where it ended. After a dozen millennia of underpopulation, the Ka'het arrival forced them to retreat. As one of the few remaining planets capable of sustaining farming this was their last stronghold, falling only centuries after the invasion begun.`

Reviewer 2: ```suggestion
		description `	This was the Builder homeworld, both the place where their journey began and where it ended. After a dozen millennia of underpopulation, the Ka'het arrival forced them to retreat. As one of the few remaining planets capable of sustaining farming, this was their last stronghold, falling centuries after the invasion begun.`
```
`only` implies a shortness to the time elapsed, which seems to contradict `centuries` to me. Even if it is a short period of time in the "grand scheme of things," if this world survives much longer than any other Builder world, then relatively speaking it's not really that "short" a time.

	planet Veliante

> 		description `	Soon after the beginning the invasion, an attack directed towards some sort of advanced nuclear facility started a meltdown; the fallout quickly spread across most of the planet, leaving all Builder settlements uninhabitable.`

Reviewer 2: ```suggestion
		description `	Soon after the beginning the invasion, an attack on some sort of advanced nuclear facility started a meltdown; the fallout quickly spread across most of the planet, leaving all Builder settlements uninhabitable.`
```
